packages/api/app/Models/TechnologyOrder.js

```diff
@@ -14,6 +14,44 @@ class TechnologyOrder extends Model {
 	user() {
 		return this.belongsTo('App/Models/User');
 	}
+
+	/**
+	 * Runs the technology order query with the provided filters.
+	 *
+	 * @param {object} query The query object.
+	 * @param {object} request The request object.
+	 * @returns {object}
+	 */
+	static async scopeWithFilters(query, request) {
+		const filters = request.only([
+			'status',
+			'unit_value',
+			'responsible',
+			'buyer',
+			'dateStart',
+			'dateEnd',
+		]);
+		if (filters.status) {
+			const statusList = filters.status ? filters.status.split(',') : [];
+			if (statusList && statusList.length) {
+				query.whereIn('status', statusList);
+			}
+		}
+
+		if (filters.unit_value) {
+			query.where('unit_value', filters.unit_value);
+		}
+
+		if (filters.buyer) {
+			query.where('user_id', filters.buyer);
+		}
+
+		const today = new Date();
+		const initialDate = new Date('2019-01-01 00:00:00');
```

Comment:
Qual o objetivo disso? 🤔
Se for pegar o primeiro dia do ano passado, melhor utilizar o `dayjs` pra manejar isso
```js
const initialDate = dayjs().subtract(1, 'year').startOf('year');
```
Mas acho que não era esse o objetivo 😅

Reply:
Aqui uma sugestão é bem vinda.
Isso ai é a data inicial quando o usuário filtrar apenas pela data final.

Ex: Liste todos os pedidos até hoje

Reply:
Acho importante resolver isso por agora. Imagino que comumente não passarão esse filtro, então, parecerá que temos um bug pois não mostrará as orders mais antigas

Reply:
@NaylsonFerreira

Reply:
Está funcionando, não haverá ordens no sistema anteriores a essa data  **2019-01-01 00:00:00**